server/utils/serviceProviderCSVGenerator.test.ts

```diff
@@ -0,0 +1,53 @@
+import ServiceProviderCSVGenerator from './serviceProviderCSVGenerator'
+
+describe(ServiceProviderCSVGenerator, () => {
+  describe('.generate', () => {
+    describe('with valid ServiceProviderReportReferrals', () => {
+      it('returns a CSV-formatted string', () => {
+        const reportingResponse = [
+          {
+            referralLink: 'https://refer-and-monitor.com/referral/c9f9e22f-ddd9-422a-a9af-76fc02b18b38',
+            referralRef: 'SM1973AC',
+            referralId: 'c9f9e22f-ddd9-422a-a9af-76fc02b18b38',
+            contractId: 'thing',
+            organisationId: 'XYZ5678',
+            referringOfficerEmail: 'joe.probation@example.com',
+            caseworkerId: 'liane.supplier@example.com',
+            serviceUserCRN: 'X017844',
+            dateReferralReceived: '2021-06-27T14:49:01+01:00',
+            dateSAABooked: '2021-06-27T14:49:01+01:00',
+            dateSAAAttended: '2021-06-27T14:49:01+01:00',
+            dateFirstActionPlanSubmitted: '2021-06-27T14:49:01+01:00',
+            dateOfFirstActionPlanApproval: '2021-06-27T14:49:01+01:00',
+            dateOfFirstAttendedSession: '2021-06-27T14:49:01+01:00',
+            outcomesToBeAchievedCount: 8,
+            outcomesAchieved: 5.5,
+            countOfSessionsExpected: 10,
+            countOfSessionsAttended: 6,
+            endRequestedByPPAt: '2021-06-27T14:49:01+01:00',
+            endRequestedByPPReason: 'REC',
+            dateEOSRSubmitted: '2021-06-27T14:49:01+01:00',
+            concludedAt: '2021-06-27T14:49:01+01:00',
+          },
+        ]
+
+        expect(ServiceProviderCSVGenerator.generate(reportingResponse)).toContain(
+          'referral_link,referral_ref,referral_id,contract_id,organisation_id,referring_officer_email,caseworker_id,service_user_crn,date_referral_received,date_saa_booked,date_saa_attended,date_first_action_plan_submitted,date_of_first_action_plan_approval,date_of_first_attended_session,outcomes_to_be_achieved_count,outcomes_achieved,count_of_sessions_expected,count_of_sessions_attended,end_requested_by_pp_at,end_requested_by_pp_reason,date_eosr_submitted,concluded_at'
+        )
+        expect(ServiceProviderCSVGenerator.generate(reportingResponse)).toContain(
+          'https://refer-and-monitor.com/referral/c9f9e22f-ddd9-422a-a9af-76fc02b18b38,SM1973AC,c9f9e22f-ddd9-422a-a9af-76fc02b18b38,thing,XYZ5678,joe.probation@example.com,liane.supplier@example.com,X017844,2021-06-27T14:49:01+01:00,2021-06-27T14:49:01+01:00,2021-06-27T14:49:01+01:00,2021-06-27T14:49:01+01:00,2021-06-27T14:49:01+01:00,2021-06-27T14:49:01+01:00,8,5.5,10,6,2021-06-27T14:49:01+01:00,REC,2021-06-27T14:49:01+01:00,2021-06-27T14:49:01+01:00'
+        )
+      })
+    })
+  })
+
+  describe('.createFilename', () => {
+    it('returns a filename including the date the report is generated', () => {
+      const mockDate = new Date('2021-06-28')
```

Comment:
Can we be more specific when constructing this date, so it's clear exactly which moment in time this is referring to? e.g. `new Date('2021-06-28T10:00:00Z')`